proposals/graduation/falco.md

```diff
@@ -0,0 +1,78 @@
+
+On January 8th, 2020 Falco was [promoted to the Incubating stage](https://www.cncf.io/blog/2020/01/08/toc-votes-to-move-falco-into-cncf-incubator/). Since then, the Falco community, users and maintainers continued to actively invest in Falco, fueling sustained growth in terms of adopters, governance, contributions, features and integrations with other CNCF projects.
+
+At this stage, we believe that Falco is ready for the next natural step: Graduation.
+
+Key highlights as of April 19th, 2021:
+
+- Github stars: 4.200+
+- Core repository contributors: 100+
+- Forks: 660+
+- Releases: 110
+- Slack channel members: 750+
+- Downloads:
+  - 24M+ Docker hub pulls (+570% since Incubation)
+  - 30M+ driver downloads from our CDN (+680% since Incubation)
+	- Since Falco compiles a driver for the host when it starts (and fallbacks to download a prebuilt one) this is a very conservative proxy of how many times Falco is installed
+- Top 3 media mentioned Incubated CNCF project for last 3 months (January, February, March 2021)
+- Keynote mention at Kubecon+CloudNativeCon 2020 -- Constance Caramolis ([youtube](https://www.youtube.com/watch?v=uNpZA5fz5o8))
+- End user talk at Kubecon 2020 -- Shane Lawrence, Shopify ([youtube](https://www.youtube.com/watch?v=rBqBrYESryY))
+- End user talk at Kubecon 2020 -- Natch Ruengsakulrach & Eric Hollis, MathWorks ([slides](https://static.sched.com/hosted_files/kccncna20/aa/KubeCon_NA_Virtual_2020-Cyber_Kill_Chain_Falco.pdf))
+- Inclusion in CNCF CKS certification
+- Several vendors have built commercial offerings on top of Falco (e.g. [GitLab](https://about.gitlab.com/blog/2020/08/18/how-gitlab-can-help-you-secure-your-cloud-native-applications/), [SumoLogic](https://www.sumologic.com/solutions/kubernetes/))
+- The Falco organization is evolving from a single tool into an ecosystem of projects, some of which are gaining substantial traction (e.g. [falcosidekick](https://github.com/falcosecurity/falcosidekick): 390K+ Docker hub pulls, [falco-exporter](https://github.com/falcosecurity/falco-exporter): 260K+ Docker hub pulls)
+
+## Graduation State Criteria
+
+### ✅ Have committers from at least two organizations.
+
+Since incubation, the amount of contributions coming from organizations other than Falco original creator (Sysdig) has grown steadily. During the last year, around 45% of the contributions came from a diverse group of committers that includes Innoteam, Amazon, Samsung, IBM, Mercari, Red Hat and many individual contributors. More details are available [here](https://falco.devstats.cncf.io/d/5/companies-table?orgId=1&var-period_name=Last%20year&var-metric=contributions).
+
+[Maintainers](https://github.com/falcosecurity/.github/blob/master/maintainers.yaml) of [Falco projects](https://github.com/falcosecurity) also come from a wide set of companies (Amazon, IBM, VMWare, RedHat, Sysdig, Qonto, move:elevator GmbH, Mercari, Timber).
```

Comment:
SIG-Security would like to raise a concern regarding the vendor independence of the Falco project.  The[ Incubation Due Diligence document](https://docs.google.com/document/d/1TJCzW8dQ6858lw2UNY-H5LMnvEd4GzwjuOcDInimeyA/edit?ts=5dacfd96#heading=h.378jkvcve1nq) shows many items that are indicated to be resolved prior to graduation.  Reviewing the existing codebase highlights several areas without independence but also [reflect efforts of that transition](https://github.com/falcosecurity/driverkit/commit/a4924530ba68673d661db9652595768e66ba9512).  This in turn causes confusion to the community as it is not clear what the scope of Falco is.  Organizations may not simply deploy Falco as is, rather they must also deploy sysdig to reap the value of the project.

If the separation were to be completed, the existing security audit of 2019 may no longer be relevant to the project as significant changes may have occurred.  The scheduled security audit for later this year would be beneficial, provided the separation were completed.

CC @lizrice & @justincormack

Reply:
Hello Emily!

I'd like to answer your comment by summarizing the discussion (and the plan we came up with) during our call last week.

To recap:

1. The **current Falco prebuilt drivers** - those installed when using the DEB, RPM, and tarball Falco packages - **are already built from the contributed libs repository** thanks to [driverkit](https://github.com/falcosecurity/driverkit/search?q=DownloadBaseURL) and our [Open Falco Infra](https://prow.falco.org) (Prow) (see [build-drivers](https://github.com/falcosecurity/test-infra/tree/master/images/build-drivers), its [ProwJobs definitions](https://github.com/falcosecurity/test-infra/blob/master/config/jobs/build-drivers/build-drivers.yaml), and an [example run](https://prow.falco.org/log?job=build-drivers-debian-periodic&id=1387602592925749248))
2. There are **2 ongoing PRs to complete the refactoring of the CMakeLists.txt files** both [in Falco](https://github.com/falcosecurity/falco/pull/1552) and [in libs](https://github.com/falcosecurity/libs/pull/30) repository
3. Today, I've finally published 5 **GitHub Security Advisories** on `falcosecurity/libs` **containing detailed information about all the finding of the security audit** done in July 2019, and the relative fixes
      - You can find them [here](https://github.com/falcosecurity/libs/security/advisories?state=published)
4. The findings (and the fixes) of the **security audit** regarding **Falco** only have been **already published** as [Github Security Advisories](https://github.com/falcosecurity/falco/security/advisories)
5. The `falcosecurity/libs` code **doesn't present substantial changes** in the API or in the behavior to the code base that underwent a security audit in July 2019
  The only differences are due to:
      - Fixes to vulnerabilities found during the security audit
      - Fixes on the eBPF driver (mostly to make the eBPF verifier happy on newer Linux kernels)
      - Fixes regarding the `bpf_probe_read_str()` usage
      - Fixes to the handling of syscalls getting relative or absolute file paths
      - Fixes to properly use newer libcurl features
      - Memory leaks fixes
      - Support for tracing the `renameat2` syscall
      - Support for tracing the `fchmodat`, `fchmod`, `chmod` syscalls
      - Support to fetch Kubernetes liveness/readiness probe from related containers (when not available in the container info labels)
      - Async DNS resolution using `c-ares`
      - The userspace API (an experimental API - see pdig driver, or `-u` Falco flag)
      - Decoupling and refactoring code (cosmetic changes) to let the libs contribution happen
      - Refactoring of the CMakeLists.txt files to let the libs contribution happen
      
      We can prepare a deck/presentation to present the libs repository to the SIG-Security, if requested

Roadmap: as you may notice, there's a lot already done to complete the transition, and there's active ongoing work (**point 2**) to complete it that will be complete very soon.

Feel free to reach out if you have any questions. :)